querydsl-jpa/src/main/java/com/querydsl/jpa/hibernate/SessionHolder.java

```diff
@@ -24,6 +27,30 @@
  */
 public interface SessionHolder {
 
+    /**
+     * Workaround for Hibernate 5.2's return type changes.
+     *
+     * Because we can't have the return type in source code form, we have to use reflection.
+     * This means that the method lookup happens just by name.
```

Comment:
@timowest Because the method `org.hibernate.Query.createQuery(Ljava/lang/String)Lorg/hibernate/SQLQuery` (and others) doesn't exist anymore, we have to invoke it by name.
Although on closer inspection I see that I missed a bunch more.
```
java.lang.NoSuchMethodError: org.hibernate.SQLQuery.addScalar(Ljava/lang/String;)Lorg/hibernate/SQLQuery;
java.lang.NoSuchMethodError: org.hibernate.Session.close()Ljava/sql/Connection;
And probably others
```
I now notice what kind of mess they made, I think the only way is to either ditch pre 5.2 or to figure out some additional maven artifact for pre 5.2.

Reply:
+1 for a new release with 5.2 support (maybe even QDSL5?) :)